mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -103,13 +111,22 @@ public MaBookmark getBookmark(Long callingNumber, Long callId) {
     @Override
     public void setBookmark(MaBookmark saveBookmark) {
 
-        List<Bookmark> existing = bookmarkDataService.findBookmarksForUser(saveBookmark.getCallingNumber().toString());
+        String callingNumber = saveBookmark.getCallingNumber().toString();
+        List<Bookmark> existing = bookmarkDataService.findBookmarksForUser(callingNumber);
 
         if (CollectionUtils.isEmpty(existing)) {
             // if no bookmarks exist for user
+            LOGGER.info("No bookmarks found for user " + callingNumber);
             bookmarkDataService.create(setBookmarkProperties(saveBookmark, new Bookmark()));
         } else {
-            // we found a list (usually only 1) and update it
+            // error check
+            if (existing.size() > 1) {
+                LOGGER.error("Found more than 1 bookmark for calling number. This should never be possible.");
+                LOGGER.error("Contact dev team about calling number: " + callingNumber);
+            }
+
+            // update the first bookmark
```

Comment:
Given the above check (of only exactly one), shouldn't that read `//update the bookmark`?